lib/web/apiserver.go

```diff
@@ -3376,16 +3382,28 @@ func (h *Handler) podConnect(
 	ws *websocket.Conn,
 ) (interface{}, error) {
 	q := r.URL.Query()
-	params := q.Get("params")
-	if params == "" {
+	if q.Get("params") == "" {
 		return nil, trace.BadParameter("missing params")
 	}
-
-	var execReq PodExecRequest
-	if err := json.Unmarshal([]byte(params), &execReq); err != nil {
+	var params podConnectParams
+	if err := json.Unmarshal([]byte(q.Get("params")), &params); err != nil {
 		return nil, trace.Wrap(err)
 	}
 
+	execReq, err := readPodExecRequestFromWS(ws)
+	if err != nil {
+		if errors.Is(err, io.EOF) || errors.Is(err, net.ErrClosed) || terminal.IsOKWebsocketCloseError(trace.Unwrap(err)) {
+			return nil, nil
+		}
+		var netError net.Error
+		if errors.As(trace.Unwrap(err), &netError) && netError.Timeout() {
+			return nil, trace.BadParameter("timed out waiting for pod exec request data on websocket connection")
+		}
+
+		return nil, err
```

Comment:
Handler's `WithClusterAuthWebSocket`, which podConnect uses, writes returned error to the web socket.

Reply:
FWIW, I had the same experience, but I assumed it's expected. An error message there would be nice though.

Reply:
Turned out that even though `WithClusterAuthWebSocket` was writing an error to the websocket, it was doing that incorrectly, missing version field
![image](https://github.com/gravitational/teleport/assets/964862/b5a52dec-f9c8-42b3-abee-cda63108ab1b)

But fixing that doesn't help, because client side tty doesn't actually know what to do with an error, it doesn't have a handler for that type of message, so I guess that never worked - it also doesn't work for the ssh nodes connections, if you return an error in the handler, it also just displays "disconnected"

![image](https://github.com/gravitational/teleport/assets/964862/746d244e-845a-4b37-b3d8-d200d40e2a59)

We can fix it, but since it's a general problem I would like it to be a separate PR (I'll open an issue).

Reply:
Opened: https://github.com/gravitational/teleport/issues/42626